revoke/revoke.go

```diff
@@ -123,69 +154,145 @@ func getIssuer(cert *x509.Certificate) *x509.Certificate {
 	}
 
 	return issuer
-
 }
 
-// check a cert against a specific CRL. Returns the same bool pair
-// as revCheck.
-func certIsRevokedCRL(cert *x509.Certificate, url string) (revoked, ok bool) {
-	crl, ok := CRLSet[url]
+// checks whether CRL in memory is valid
+func (r *Revoke) isInMemoryCRLValid(key string) bool {
+	crl, ok := r.CRLSet[key]
 	if ok && crl == nil {
 		ok = false
-		delete(CRLSet, url)
+		r.CRLSetLck.Lock()
```

Comment:
This is already not a main focus. Main focus is to make this lib to be more suitable for external apps.